compiler/ballerina-compiler-api/src/main/java/org/ballerina/compiler/impl/BallerinaSemanticModel.java

```diff
@@ -71,12 +80,19 @@ public BallerinaSemanticModel(BLangCompilationUnit compilationUnit, BLangPackage
         Map<Name, List<Scope.ScopeEntry>> scopeSymbols =
                 symbolResolver.getAllVisibleInScopeSymbols(lookupVisitor.lookUp(this.compilationUnit));
 
-        for (Entry<Name, List<ScopeEntry>> entry : scopeSymbols.entrySet()) {
+        DiagnosticPos cursorPos = new DiagnosticPos(new BDiagnosticSource(bLangPackage.packageID, compilationUnit.name),
```

Comment:
is it possible to handle the cursor position without converting to a diagnostic position?

Reply:
Will see. Did this conversion to make it easier to compare the user provided line position with the positions in `BSymbol`